label-application-rule-config-service-impl/src/main/java/org/hypertrace/label/application/rule/config/service/LabelApplicationRuleConfigServiceImpl.java

```diff
@@ -87,25 +89,32 @@ public void getLabelApplicationRules(
     try {
       RequestContext requestContext = RequestContext.CURRENT.get();
       this.requestValidator.validateOrThrow(requestContext, request);
-      List<LabelApplicationRule> labelApplicationRules =
-          this.labelApplicationRuleStore.getAllObjects(requestContext).stream()
-              .map(ConfigObject::getData)
-              .collect(Collectors.toUnmodifiableList());
-      Set<String> labelApplicationRuleIds =
-          labelApplicationRules.stream()
-              .map(LabelApplicationRule::getId)
-              .collect(Collectors.toUnmodifiableSet());
-      Set<String> deletedSystemLabelApplicationRuleIds =
-          getDeletedSystemLabelApplicationRuleIds(requestContext);
-      List<LabelApplicationRule> filteredSystemLabelApplicationRules =
-          this.labelApplicationRuleConfig.getSystemLabelApplicationRules().stream()
-              .filter(rule -> !labelApplicationRuleIds.contains(rule.getId()))
-              .filter(rule -> !deletedSystemLabelApplicationRuleIds.contains(rule.getId()))
-              .collect(Collectors.toUnmodifiableList());
+      List<LabelApplicationRule> allLabelApplicationRules =
+          getAllLabelApplicationRules(requestContext);
       responseObserver.onNext(
           GetLabelApplicationRulesResponse.newBuilder()
-              .addAllLabelApplicationRules(labelApplicationRules)
-              .addAllLabelApplicationRules(filteredSystemLabelApplicationRules)
+              .addAllLabelApplicationRules(allLabelApplicationRules)
+              .build());
+      responseObserver.onCompleted();
+    } catch (Exception e) {
+      responseObserver.onError(e);
+    }
+  }
+
+  @Override
+  public void getLabelApplicationRule(
+      GetLabelApplicationRuleRequest request,
+      StreamObserver<GetLabelApplicationRuleResponse> responseObserver) {
+    try {
+      RequestContext requestContext = RequestContext.CURRENT.get();
+      LabelApplicationRule labelApplicationRule =
+          getAllLabelApplicationRules(requestContext).stream()
```

Comment:
We don't need to fetch all label application rules in this case. Use getData(...) method in store  impl.